File: crates/deslop/tests/fixtures/incremental-multilang/src/LedgerAlpha.cs
// LedgerAlpha.cs — the canonical copy of the C# reconciliation routine.

namespace Ledger;

public sealed class LedgerReconciler
{
    public static long ReconcileEntries(long[] entries, long floor)
    {
        long balance = 0;
        foreach (var entry in entries)
        {
            if (entry > floor)
            {
                balance += entry * 2;
            }
            else
            {
                balance -= entry / 2;
            }
        }
        return balance;
    }
}

File: crates/deslop/tests/fixtures/incremental-multilang/src/ledger_alpha.rs
// ledger_alpha.rs — the canonical copy of the Rust reconciliation routine.

pub const ALPHA_LEDGER_TAG: &str = "alpha";

pub fn reconcile_entries(entries: &[i64], floor: i64) -> i64 {
    let mut balance = 0;
    for entry in entries {
        if *entry > floor {
            balance += entry * 2;
        } else {
            balance -= entry / 2;
        }
    }
    balance
}
